### DISCUSSION TITLE
Using alamode with phonopy

### DISCUSSION THREAD

**Discussion by Xuntao1** [ttadano/alamode#154] [Q&A] [answered]
Dear users and developers,

In the phonopy and phono3py there is an options to use (--alm) to generate IFCs. I searched online but could not figure out the best way to use this. If you can give me some hints about using this that will be very helpful.

Best,
Xun

**Comment by triphysics** [comment, marked as answer]
Do you mean to use ALM as a force constants calculator  (https://github.com/ttadano/ALM)?  

If you want to use cli you can do it the following way:
1. Follow the procedure to generate FORCE_SETS.
2. Once this file is created you can use the following to invoke the alm (assuming your supercell size is 2x2x2)

``
phonopy --alm --dim 2 2 2 -c POSCAR -v
``
If you want to write the force constants  use the following

``
phonopy --alm --dim 2 2 2 -c POSCAR -v --writefc
``

also have a look at https://github.com/ttadano/ALM/issues/33

Note that there is relatively new feature in phonopy which also work with random-displacements. Take a look at :
https://phonopy.github.io/phonopy/setting-tags.html#random-displacements

Hope this helps.

Best,
Trib

**Reply by Xuntao1** [reply]
Thank you for the pointers. This is extactly what I wanted to do.

Best,
Xun
